docs/grammar/BallerinaParser.g4

```diff
@@ -423,7 +422,15 @@ actionInvocationStatement
     ;
 
 transactionStatement
-    :   TRANSACTION LEFT_BRACE statement* RIGHT_BRACE transactionHandlers
+    :   TRANSACTION (WITH transactionPropertyInitExpressionList)? LEFT_BRACE statement* RIGHT_BRACE transactionHandlers
+    ;
+
+transactionPropertyInitExpression
+    : retriesStatement
+    ;
+
+transactionPropertyInitExpressionList
+    : transactionPropertyInitExpression (COMMA transactionPropertyInitExpression)*
```

Comment:
Fixed this by adding semantic validation to check whether retry expression is already initialized with 1e3b526e58cd325f0cb13d632c09ddf6b0e89a3f